### PR TITLE
Add render API wrapper

### PR DESCRIPTION
**Pull request by sztomi**
I started working on a wrapper for the rendering API. I've never done anything like this, so it's probably full of questionable things... let me know if this is at least in a direction you would be open to merging, and if you have specific concerns. 

#9 #11

### CODE REVIEW

File: src/mpv/render.rs [hunk 1]
```diff
@@ -15,3 +15,273 @@
...
+        }
+        // the raw array must end with type = 0
+        raw_params.push(mpv_render_param {
+            type_: 0,
+            data: ptr::null_mut(),
+        });
+
+        unsafe {
+            let raw_array = Box::into_raw(raw_params.into_boxed_slice()) as *mut mpv_render_param;
+            let ctx: mpv_render_context = MaybeUninit::uninit().assume_init();
```
Comment (anlumo) [inline src/mpv/render.rs:237]: That is undefined behavior!

**Comment by anlumo** [comment]
I did some cleanup in [my fork here](https://github.com/anlumo/libmpv-rs/). I have yet to test it, though, but it does compile without any warnings or clippy complaints.

**Comment by sztomi** [comment]
@anlumo thanks a lot for looking into this. I know I was probably doing some sketchy things, I'm pretty new to Rust. I'm also not sure about that list of pointers that `drop` frees. But if I remember correctly I couldn't find any guidance in the mpv docs about the expected lifetimes of these variables so I figured that it's best to keep them around.

**Comment by anlumo** [comment]
See #22. I'm able to watch video via OpenGL!

**Comment by sztomi** [comment]
@anlumo awesome! I'll close this one then. Thanks a bunch for keeping at it.

**Pull request closed without merging**